crypto/zkp/schemes/cl/user_credential_receiver.go

```diff
@@ -165,53 +165,69 @@ func (r *UserCredentialReceiver) getCommitmentsOfAttrsProof(challenge *big.Int)
 	return commitmentsOfAttrsProofs
 }
 
-// GetCredentialRequest computes U and returns:
+type CredentialRequest struct {
+	NymProof                 *dlogproofs.SchnorrProof
+	U                        *big.Int
+	UProof                   *qrspecialrsaproofs.RepresentationProof
+	CommitmentsOfAttrsProofs []*commitmentzkp.DFOpeningProof
+}
+
+func NewCredentialRequest(nymProof *dlogproofs.SchnorrProof, U *big.Int,
+	UProof *qrspecialrsaproofs.RepresentationProof, commitmentsOfAttrsProofs []*commitmentzkp.DFOpeningProof) *CredentialRequest {
+	return &CredentialRequest{
+		NymProof: nymProof,
+		U:        U,
+		UProof:   UProof,
+		CommitmentsOfAttrsProofs: commitmentsOfAttrsProofs,
+	}
+}
+
+// GetCredentialRequest computes U and returns CredentialRequest which contains:
 // - proof data for proving that nym was properly generated,
 // - U and proof data that U was properly generated,
 // - proof data for proving the knowledge of opening for commitments of attributes (for those attributes
 // for which the committed value is known).
-func (r *UserCredentialReceiver) GetCredentialRequest(nym *Nym, nonceOrg *big.Int) (*dlogproofs.SchnorrProof,
-	*big.Int, *qrspecialrsaproofs.RepresentationProof, []*commitmentzkp.DFOpeningProof, error) {
+func (r *UserCredentialReceiver) GetCredentialRequest(nym *Nym, nonceOrg *big.Int) (*CredentialRequest, error) {
 	r.setU()
 	nymProofRandomData, UProofRandomData, err := r.getCredentialRequestProofRandomfData(nym.Id)
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, err
 	}
 
 	challenge := r.GetChallenge(nym.Commitment, nonceOrg)
 	nymProofData, UProofData, err := r.getCredentialRequestProofData(challenge)
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, err
 	}
 
 	commitmentsOfAttrsProofs := r.getCommitmentsOfAttrsProof(challenge)
 
-	return dlogproofs.NewSchnorrProof(nymProofRandomData, challenge, nymProofData),
+	return NewCredentialRequest(dlogproofs.NewSchnorrProof(nymProofRandomData, challenge, nymProofData),
 		r.U, qrspecialrsaproofs.NewRepresentationProof(UProofRandomData, challenge, UProofData),
-		commitmentsOfAttrsProofs, nil
+		commitmentsOfAttrsProofs), nil
 }
 
 func (r *UserCredentialReceiver) GetNonce() *big.Int {
 	b := new(big.Int).Exp(big.NewInt(2), big.NewInt(int64(r.User.Params.SecParam)), nil)
 	return common.GetRandomInt(b)
 }
 
-func (r *UserCredentialReceiver) VerifyCredential(A, e, v11 *big.Int,
+func (r *UserCredentialReceiver) VerifyCredential(credential *Credential,
```

Comment:
Please use `cred` or maybe `c` as  a parameter name instead of `credential`. Go favors shorter parameter names,  especially for variables with small scopes.